newIDE/app/src/AssetStore/ProductLicense/ProductLicenseOptions.js

```diff
@@ -0,0 +1,288 @@
+import { selectMessageByLocale } from '../../Utils/i18n/MessageByLocale';
+import PlaceholderError from '../../UI/PlaceholderError';
+import { Trans } from '@lingui/macro';
+import {
+  type PrivateGameTemplateListingData,
+  type PrivateAssetPackListingData,
+} from '../../Utils/GDevelopServices/Shop';
+import { createStyles, makeStyles } from '@material-ui/core';
+import Radio from '@material-ui/core/Radio';
+import ButtonBase from '@material-ui/core/ButtonBase';
+import { shouldValidate } from '../../UI/KeyboardShortcuts/InteractionKeys';
+import { ColumnStackLayout, LineStackLayout } from '../../UI/Layout';
+import { Column, Line } from '../../UI/Grid';
+import { formatProductPrice } from '../ProductPriceTag';
+import CheckCircle from '../../UI/CustomSvgIcons/CheckCircle';
+
+const styles = {
+  buttonBase: {
+    height: '100%',
+    width: '100%',
+    borderRadius: 8,
+    padding: 8,
+    cursor: 'default',
+    overflow: 'hidden',
+    boxSizing: 'border-box',
+  },
+  contentWrapper: {
+    height: '100%',
+    width: '100%',
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  radio: {
+    padding: 0,
+  },
+  descriptionContainer: {
+    paddingLeft: 32,
+  },
+};
+
+const ProductLicenseOptionContent = ({
+  selected,
+  onClick,
+  name,
+  description,
+  formattedPrice,
+  isLicenseOwned,
+}: {|
+  selected: boolean,
+  onClick?: () => void,
+  name: string,
+  description: string,
+  formattedPrice?: string,
+  isLicenseOwned: boolean,
+|}) => (
+  <ColumnStackLayout noMargin>
+    <LineStackLayout noMargin alignItems="center">
+      <Column noMargin>
+        {isLicenseOwned ? (
+          <CheckCircle />
+        ) : onClick ? (
+          <Radio
+            color="secondary"
+            checked={selected}
+            onChange={onClick}
+            style={styles.radio}
+          />
+        ) : null}
+      </Column>
+      <LineStackLayout justifyContent="space-between" noMargin expand>
+        <Text noMargin size="sub-title">
+          {name}
+        </Text>
+        {!isLicenseOwned && formattedPrice && (
+          <Text noMargin>{formattedPrice}</Text>
+        )}
+      </LineStackLayout>
+    </LineStackLayout>
+    <Line expand noMargin>
+      <div style={styles.descriptionContainer}>
+        <Text noMargin align="left">
+          {description}
+        </Text>
+      </div>
+    </Line>
+  </ColumnStackLayout>
+);
+
+// Styles to give the impression of pressing an element.
+const useStylesForButtonBase = (selected: boolean) =>
+  makeStyles(theme =>
+    createStyles({
+      root: {
+        outline: selected
+          ? `2px solid ${theme.palette.secondary.dark}`
+          : `1px solid ${theme.palette.text.disabled}`,
+        '&:focus': {
+          backgroundColor: theme.palette.action.hover,
+        },
+        '&:hover': {
+          backgroundColor: theme.palette.action.hover,
+        },
+      },
+    })
+  )();
+
+const ProductLicenseOptionButton = ({
```

Comment:
Would almost be good to have this in UI folder. I'm a bit uneasy everytime I see a material-ui import in a folder that is not UI.